adam-core/src/main/scala/org/bdgenomics/adam/rdd/sequence/SliceRDD.scala

```diff
@@ -0,0 +1,232 @@
+import org.bdgenomics.adam.rdd.{
+  AvroGenomicRDD,
+  FileMerger,
+  JavaSaveArgs
+}
+import org.bdgenomics.adam.serialization.AvroSerializer
+import org.bdgenomics.formats.avro.{
+  Sequence,
+  Slice
+}
+import org.bdgenomics.utils.interval.array.{
+  IntervalArray,
+  IntervalArraySerializer
+}
+import org.bdgenomics.utils.misc.Logging
+import scala.collection.JavaConversions._
+import scala.reflect.ClassTag
+
+private[adam] case class SliceArray(
+    array: Array[(ReferenceRegion, Slice)],
+    maxIntervalWidth: Long) extends IntervalArray[ReferenceRegion, Slice] {
+
+  def duplicate(): IntervalArray[ReferenceRegion, Slice] = {
+    copy()
+  }
+
+  protected def replace(arr: Array[(ReferenceRegion, Slice)],
+                        maxWidth: Long): IntervalArray[ReferenceRegion, Slice] = {
+    SliceArray(arr, maxWidth)
+  }
+}
+
+private[adam] class SliceArraySerializer extends IntervalArraySerializer[ReferenceRegion, Slice, SliceArray] {
+
+  protected val kSerializer = new ReferenceRegionSerializer
+  protected val tSerializer = new AvroSerializer[Slice]
+
+  protected def builder(arr: Array[(ReferenceRegion, Slice)],
+                        maxIntervalWidth: Long): SliceArray = {
+    SliceArray(arr, maxIntervalWidth)
+  }
+}
+
+object SliceRDD {
+
+  /**
+   * Helper function for building a SliceRDD when no
+   * sequence dictionary is available.
+   *
+   * @param rdd Underlying RDD. We recompute the sequence dictionary from
+   *   this RDD.
+   * @return Returns a new SliceRDD.
+   */
+  def apply(rdd: RDD[Slice]): SliceRDD = {
+
+    // get sequence dictionary
+    val sd = new SequenceDictionary(rdd.flatMap(slice => {
+      if (slice.getName != null && slice.getLength != null) {
+        Some(SequenceRecord(slice.getName, slice.getLength))
+      } else {
+        None
+      }
+    }).distinct
+      .collect
+      .toVector)
+
+    SliceRDD(rdd, sd)
+  }
+}
+
+/**
+ * A GenomicRDD that wraps Slice data.
+ *
+ * @param rdd An RDD of genomic Slices.
+ * @param sequences The reference genome this data is aligned to.
+ */
+case class SliceRDD(rdd: RDD[Slice],
+                    sequences: SequenceDictionary) extends AvroGenomicRDD[Slice, SliceRDD] with Logging {
+
+  protected def buildTree(rdd: RDD[(ReferenceRegion, Slice)])(
+    implicit tTag: ClassTag[Slice]): IntervalArray[ReferenceRegion, Slice] = {
+    IntervalArray(rdd, SliceArray.apply(_, _))
+  }
+
+  /**
+   * Save slices as Parquet or FASTA.
+   *
+   * If filename ends in .fa or .fasta, saves as FASTA. If not, saves slices
+   * to Parquet. Defaults to 60 character line length, if saving to FASTA.
+   *
+   * @param filePath Path to save files to.
+   * @param asSingleFile If true, saves output as a single file.
+   */
+  def save(filePath: java.lang.String, asSingleFile: java.lang.Boolean) {
+    if (filePath.endsWith(".fa") || filePath.endsWith(".fasta")) {
+      saveAsFasta(filePath, asSingleFile = asSingleFile)
+    } else {
+      if (asSingleFile) {
```

Comment:
A bit silly, but perhaps we can move this to `GenomicRDD` as a protected method `def warnIfSingleFile(asSingleFile: Boolean): Unit`?